### PR TITLE
Add pipeline for ma/ma_immunizations

### LINKED ISSUE #241: fetch ma/ma_immunizations
[learn our pipeline: fetch](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#fetch)

Fetch data from: https://maimmunizations.org

Put your script in a file named: `ma/ma_immunizations/fetch.sh` (or `fetch.sh`, etc.)

Store it (without processing) in a new file created in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the fetch stage and how to get set up for development!

### Tips

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest fetch ma/ma_immunizations
    ```

### LINKED ISSUE #242: parse ma/ma_immunizations
[learn our pipeline: parse](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#parse)

Blocked by #241

Parse existing data into [`.ndjson`](http://ndjson.org/).

Read all files in the directory passed as the second argument (`sys.argv[2]`), convert them to `.ndjson`, and then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the parse stage and how to get set up for development!

### Tips

1. Fetch data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch ma/ma_immunizations
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest parse ma/ma_immunizations
    ```

### Example
[Fetched files for `md/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/md/arcgis/raw) are converted to [parsed files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/md/arcgis/parsed)

### LINKED ISSUE #243: normalize ma/ma_immunizations
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Blocked by #242

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch ma/ma_immunizations
    poetry run vaccine-feed-ingest parse ma/ma_immunizations
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize ma/ma_immunizations
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by alexkuang**
# ma/ma_immunizations

| Key Details |
|-|
| Resolves #241 |
| Resolves #242 |
| Resolves #243 |
State: ma |
Site: ma_immunizations |

## Notes
I had to make a few tweaks to the prepmod infra to get this to work.
- Added support for extra params.  The site returns empty results if that `q[...]` param is missing, for some reason.
- Added support for headers.  The site seems to have some relatively naive bot checking.  It 403's with empty user agent, but will allow anything else in, e.g. `curl` worked just fine in my local testing.  Used the URL for this github as a fill-in, but let me know if there's a more official UA / good netizen convention for this project.

## Data sample

```json
{"id": "ma_immunizations:4573", "name": "Greenfield Senior Center", "address": {"street1": "35 Pleasant Street", "street2": null, "city": "Greenfield", "state": "MA", "zip": "01301"}, "location": null, "contact": [{"contact_type": "booking", "phone": null, "website": "https://www.maimmunizations.org/appointment/en/client/registration?clinic_id=4573", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-05-12", "closes": "2021-05-12"}], "opening_hours": [{"day": "wednesday", "opens": "11:00", "closes": "18:00"}], "availability": {"drop_in": null, "appointments": true}, "inventory": [{"vaccine": "moderna", "supply_level": null}], "access": null, "parent_organization": null, "links": null, "notes": ["", "If you are signing up for a second dose, you must get the same vaccine brand as your first dose."], "active": null, "source": {"source": "ma_immunizations", "id": "4573", "fetched_from_uri": "https://www.maimmunizations.org/clinic/search", "fetched_at": "2021-05-11T11:46:24.179337", "published_at": null, "data": {"name": "Greenfield Senior Center", "date": "05/12/2021", "address": "35 Pleasant Street, Greenfield MA, 01301", "vaccines": "Moderna COVID-19 Vaccine", "ages": "Adults, Seniors", "info": "", "hours": "11:00 am - 06:00 pm", "available": "0", "special": "If you are signing up for a second dose, you must get the same vaccine brand as your first dose.", "clinic_id": "4573"}}}
{"id": "ma_immunizations:4623", "name": "Saint Vincent Hospital Vaccine Collaborative @ Worcester State University - Wellness Center", "address": {"street1": "525 Chandler Street", "street2": null, "city": "Worcester", "state": "MA", "zip": "01602"}, "location": null, "contact": [{"contact_type": "booking", "phone": null, "website": "https://www.maimmunizations.org/appointment/en/client/registration?clinic_id=4623", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-05-12", "closes": "2021-05-12"}], "opening_hours": [{"day": "wednesday", "opens": "13:30", "closes": "17:30"}], "availability": {"drop_in": null, "appointments": true}, "inventory": [{"vaccine": "pfizer_biontech", "supply_level": null}], "access": null, "parent_organization": null, "links": null, "notes": ["By Appointment Only - Arrive 15min before scheduled appointment.\n\nDo not enter the building until 5 min before your scheduled appt time. Watch the digital scoreboard for appointment time prompting.\n\nBRING PHOTO ID AND INSURANCE CARD(S).\n\nMASKS are required and must be worn properly at all times.\n\nIf receiving your 2nd dose, please bring your vaccine card with you.\n\nWear clothing that allows a clinician to easily access your upper arm. The vaccine is delivered to the deltoid muscle, the big muscle on your shoulder. Consider wearing a short-sleeved shirt, or wear a short-sleeved shirt under a sweater or jacket that can be easily removed.\n\n***We kindly request that you do not bring children to the vaccination clinic site***", "If you are signing up for a second dose, you must get the same vaccine brand as your first dose."], "active": null, "source": {"source": "ma_immunizations", "id": "4623", "fetched_from_uri": "https://www.maimmunizations.org/clinic/search", "fetched_at": "2021-05-11T11:46:24.179337", "published_at": null, "data": {"name": "Saint Vincent Hospital Vaccine Collaborative @ Worcester State University - Wellness Center", "date": "05/12/2021", "address": "525 Chandler Street, Worcester MA, 01602", "vaccines": "Pfizer-BioNTech COVID-19 Vaccine", "ages": "Adults, Seniors, Other", "info": "By Appointment Only - Arrive 15min before scheduled appointment.\n\nDo not enter the building until 5 min before your scheduled appt time. Watch the digital scoreboard for appointment time prompting.\n\nBRING PHOTO ID AND INSURANCE CARD(S).\n\nMASKS are required and must be worn properly at all times.\n\nIf receiving your 2nd dose, please bring your vaccine card with you.\n\nWear clothing that allows a clinician to easily access your upper arm. The vaccine is delivered to the deltoid muscle, the big muscle on your shoulder. Consider wearing a short-sleeved shirt, or wear a short-sleeved shirt under a sweater or jacket that can be easily removed.\n\n***We kindly request that you do not bring children to the vaccination clinic site***", "hours": "01:30 pm - 05:30 pm", "available": "592", "special": "If you are signing up for a second dose, you must get the same vaccine brand as your first dose.", "clinic_id": "4623"}}}
```

## Before Opening a PR
- [x] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [x] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

**Comment by bryanculbertson** [comment]
This looks good to me, bit I would like @eliblock to review the User Agent part.

File: vaccine_feed_ingest/runners/ma/ma_immunizations/fetch.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+state: ma
+site: ma_immunizations
+parser: prepmod
+url: https://www.maimmunizations.org
+params:
+  q[venue_search_name_or_venue_name_i_cont]: ''
+headers:
+  User-Agent: github.com/CAVaccineInventory
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/ma/ma_immunizations/fetch.yml:9]: Could you use `Vaccinebot (+https://www.vaccinatethestates.com/vaccinebot)` as the user agent?

We'll be adding a page to vaccinateTheStates.com similar to [this page on vaccinateCA.com](https://www.vaccinateca.com/vaccinebot) to provide more information for folks who look it up.

Comment (alexkuang) [reply vaccine_feed_ingest/runners/ma/ma_immunizations/fetch.yml]: Yep, done.

**Review by bryanculbertson — APPROVED** [review]
Approving and merging with the new user agent

**Pull request merged**